### PR TITLE
List of ingredients

### PR DESCRIPTION
**Pull request by natimati**

### CODE REVIEW

File: list.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+class Component {
```
Comment (janowsiany) [inline list.js:1]: move the file to src/components/currentRecipe.js
```suggestion
class CurrentRecipe {
```

File: list.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  constructor() {
+    this.container = document.getElementById("ingredients-list");
```
Comment (janowsiany) [inline list.js:3]:
```suggestion
    this.ref = document.getElementById("ingredientsList");
```
we call it `ref` in other places and we use camelCase for classnames

File: list.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  renderInput() {
+    const input = document.createElement("input");
+    input.setAttribute("type", "text");
+    input.setAttribute("placeholder", "Type ingredient and press enter");
+    this.container.appendChild(input);
+    const addListElement = (event) => {
+      if (event.key === "Enter") {
+        this.addNewListElement(input.value);
+        input.value = "";
```
Comment (janowsiany) [inline list.js:31]: does calling reset method will have the same effect?

File: index.html [hunk 4]
```diff
@@ -7,6 +7,8 @@
     <title>CodersCamp 2021 - Project 1</title>
   </head>
   <body>
-    Hello World!
+    <!-- <input type="text" placeholder="type ingredient"> -->
+    <div id="ingredients-list"></div>
+    <script src="list.js" defer></script>
```
Comment (janowsiany) [inline index.html:12]: For now lets remove changes to that file

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: list.js (hunk 3 above)
Comment (natimati) [reply list.js:31]: I checked but input wasn't cleared after using reset method

File: list.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  addNewListElement(ingredientName) {
+    const listElement = document.createElement("li");
+    listElement.innerText = ingredientName;
+    listElement.id = ingredientName;
+    listElement.classList.add("listElement");
+    this.list.appendChild(listElement);
+    const removeButton = document.createElement("button");
+    removeButton.setAttribute("type", "button");
+    // removeButton.innerText = "remove ingredient"; //TODO: use icon
```
Comment (janowsiany) [inline list.js:14]: can we removve it please

File: list.js (hunk 2 above)
Comment (janowsiany) [reply list.js:3]: this is still to be fixed

File: list.js (hunk 1 above)
Comment (janowsiany) [reply list.js:1]: this is still to be fixed

File: list.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+  render(props) {
+    this.renderInput();
+    this.renderList(props.currentRecipe.ingredients);
+    this.renderSubmitButton();
+  }
+}
+
+const newComponent = new Component();
+newComponent.render({ currentRecipe: { ingredients: ["orange", "potato"] } });
```
Comment (janowsiany) [inline list.js:69]: move initialization to index.js

**Review by janowsiany — CHANGES_REQUESTED** [review]
If we already have basic layout prepared by Weronika, can you please add `ingredientsList` to index.html so it actually works?

**Comment by natimati** [comment]
@janowsiany please take a look, I fixed all issues

File: src/CurrentRecipe.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+class CurrentRecipe {
```
Comment (janowsiany) [inline src/CurrentRecipe.js:1]: please move it to correct destination https://github.com/CodersCamp2021/team-jo-project-1/pull/21#discussion_r768094668

**Review by janowsiany — CHANGES_REQUESTED** [review]

File: src/CurrentRecipe.js (hunk 7 above)
Comment (natimati) [reply src/CurrentRecipe.js]: done

**Review by janowsiany — APPROVED** [review]

**Pull request merged**